src/auth/startmanager.cpp

```diff
@@ -0,0 +1,422 @@
+        // create new user window
+        NewUserWindow newUserWindow(this);
+        connect(&newUserWindow, &NewUserWindow::cancel, this, [=](){emit cancel();});  // user pressed Cancel button in bottom right side
+        connect(&newUserWindow, &NewUserWindow::rejected, this, [=](){emit cancel();});// user pressed Cross button in top right side
+
+        // user clicked button [Try without authentication] - need to do three things:
+        connect(&newUserWindow, &NewUserWindow::unnamedUser, this, &StartManager::onUnnamedUserChoice); // assign [mUserMode], [mUserName], [mToken]
+        connect(&newUserWindow, &NewUserWindow::unnamedUser, &newUserWindow, &NewUserWindow::accept);   // terminate newUserWindow
+        connect(&newUserWindow, &NewUserWindow::unnamedUser, this, [&](){                               // show notification window
+            QEventLoop loop;
+            QMessageBox *mbox = new QMessageBox;
+            mbox->setWindowTitle(tr("Welcome to Pair Storm!"));
+            QString msg {"You entering in unauthenticated mode"};
+            mbox->setText(msg);
+            mbox->setStandardButtons(nullptr);
+            mbox->show();
+            QTimer::singleShot(mTimeOutWelcomeWindow, this, [&](){
+                    loop.exit();
+                    mbox->accept();
+                });
+            loop.exec();
+                    });
+
+        connect(&newUserWindow, &NewUserWindow::newUserToken, this, &StartManager::onNewUserToken);// during registration user typed login and token
+        connect(&newUserWindow, &NewUserWindow::newUserPasssword, this, &StartManager::onNewUserPassword);// during registration user typed login and password
+        connect(this, &StartManager::cancelNewUserWindow, &newUserWindow, &NewUserWindow::accept);// token validated (or got new token) - close newUserWindow
+
+        newUserWindow.exec();
+
+        break;
+    }
+    default:
+    {
+        break;
+    }
+    }
+
+    //  H
+    StoreConf conf(mUserName);
+    conf.restoreConFile();
+    if (mUserMode == userMode::NewUser)     //  set token to new user
+    {
+        conf.setField("token", mToken);
+        QSettings s;
+        s.setValue("token", mToken);
+    }
+}
+
+void StartManager::setPathToConfDir(const QString path)
+{
+    mPathToConfDir = path;
+}
+
+void StartManager::makeListRegisteredUsers()
+{
+    mListRegisteredUsers = UsersListMaker::usersListMake(mPathToConfDir);
+    //qDebug() << "list registered users" << mListRegisteredUsers;
+}
+
+void StartManager::validateToken()
+{
+    // GET request
+    QEventLoop loop;
+    connect(mpDownloader, &DownloaderWrapper::respond, [&](const QString &respond){
+            if (respond == "normal")    mRespondStatus = respondStatus::Normal;
+            if (respond == "error")     mRespondStatus = respondStatus::Error;
+            if (respond == "corrupted") mRespondStatus = respondStatus::Corrupted;
+            //qDebug() << "got respond" << respond;
```

Comment:
There is no need to transform string messages into status enum.
Just use common enum